### PR TITLE
[PBE-3848] add additional option over message xml compose for user blocking

### PR DESCRIPTION
**Pull request by aleksandar-apostolov**
### 🎯 Goal

Connect the User blocking API with the UI components

### 🛠 Implementation details

Added both `Event` and `MessageAction`

### 🎨 UI Changes

image

### CODE REVIEW

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/messageoptions/MessageOptions.kt [hunk 1]
```diff
@@ -145,6 +146,7 @@ public fun defaultMessageOptionsState(
     val canEditAnyMessage = ownCapabilities.contains(ChannelCapabilities.UPDATE_ANY_MESSAGE)
     val canMarkAsUnread = ownCapabilities.contains(ChannelCapabilities.READ_EVENTS)
     val canFlagMessage = ownCapabilities.contains(ChannelCapabilities.FLAG_MESSAGE)
+    val canBlockUser = true
```
Comment (kanat) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/messageoptions/MessageOptions.kt:149]: Is this always supposed to be `true`?

File: stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/ui/common/feature/messages/list/MessageListController.kt [hunk 2]
```diff
@@ -1873,6 +1875,32 @@ public class MessageListController(
...
+
+    /**
+     * Unblock a user.
+     *
+     * @param userId the id of the user that will be unblocked.
+     */
+    public fun unblockUser(userId: String) {
+        chatClient.unblockUser(userId).enqueue(onError = { error ->
+            onActionResult(error) {
+                ErrorEvent.BlockUserError(it)
```
Comment (kanat) [inline stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/ui/common/feature/messages/list/MessageListController.kt:1899]: Should it be `ErrorEvent.UnblockUserError(it)`

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/feature/messages/list/options/message/MessageOptionItemsFactory.kt [hunk 3]
```diff
@@ -120,6 +121,7 @@ public open class DefaultMessageOptionItemsFactory(
         val canEditAnyMessage = ownCapabilities.contains(ChannelCapabilities.UPDATE_ANY_MESSAGE)
         val canMarkAsUnread = ownCapabilities.contains(ChannelCapabilities.READ_EVENTS)
         val canFlagMessage = ownCapabilities.contains(ChannelCapabilities.FLAG_MESSAGE)
+        val canBlockUser = true
```
Comment (kanat) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/feature/messages/list/options/message/MessageOptionItemsFactory.kt:124]: Do we need this flag if it's always `true`?

File: stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/ui/common/feature/messages/list/MessageListController.kt (hunk 2 above)
Comment (aleksandar-apostolov) [reply stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/ui/common/feature/messages/list/MessageListController.kt:1899]: Its the same for Ban and Unban

**Review by kanat — APPROVED** [review]
LGTM

**Pull request merged**